### PR TITLE
Add RedisReporter for minitest

### PR DESCRIPTION
**Pull request by casperisfine**
Add a RedisReporter for minitest, to be able to centralize the result for the CI run.

@Shopify/pipeline for review please (Note: It's at 99% what was inside Shopify, with added test coverage.)

### CODE REVIEW

File: lib/ci/queue/redis/base.rb [hunk 1]
```diff
@@ -44,10 +44,10 @@ def wait_for_master(timeout: 10)
 
         private
 
-        attr_reader :redis
+        attr_reader :redis, :build_id
 
         def key(*args)
-          [@key, *args].join(':')
+          ["build:#{build_id}", *args].join(':')
```
Comment (wvanbergen) [inline lib/ci/queue/redis/base.rb]: `['build', build_id, *args].join(':)` ?

File: lib/minitest/reporters/redis_reporter.rb [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+
+        def record(test)
+          super
+
+          self.total_time = Minitest.clock_time - start_time
+          if test.skipped?
+            self.skips += 1
+          elsif test.error?
+            self.errors += 1
+          elsif test.failure
```
Comment (wvanbergen) [inline lib/minitest/reporters/redis_reporter.rb]: not `test.failure?` ?

File: test/fixtures/redis-runner.rb [hunk 3]
```diff
@@ -16,3 +16,5 @@
 if ARGV.first == 'retry'
   Minitest.queue = Minitest.queue.retry_queue
 end
+
+$ansi = false # Disable ANSI codes for easier testing
```
Comment (wvanbergen) [inline test/fixtures/redis-runner.rb]: https://github.com/Shopify/buildkite-runner/blob/master/test/lib/command_output_helpers.rb#L15-L17 may help here.

File: test/integration/redis_test.rb [hunk 4]
```diff
@@ -9,9 +9,38 @@ def setup
...
+        "    test/fixtures/dummy_test.rb:9:in `test_bar'",
+        "    lib/ci/queue/redis/worker.rb:32:in `poll'",
+        "",
+        "ERROR BTest#test_bar",
+        "TypeError: String can't be coerced into Fixnum",
+        "    test/fixtures/dummy_test.rb:19:in `+'",
+        "    test/fixtures/dummy_test.rb:19:in `test_bar'",
+        "",
+        "",
+      ].join("\n")
```
Comment (wvanbergen) [inline test/integration/redis_test.rb]: I prefer `<<~HEREDOC` syntax for this, if you're using Ruby 2.2 or up. If not, https://github.com/Shopify/buildkite-runner/blob/master/test/lib/command_output_helpers.rb#L60-L63 will do the same.

File: lib/ci/queue/redis/base.rb (hunk 1 above)
Comment (casperisfine) [inline lib/ci/queue/redis/base.rb]: :facepalm:

File: lib/minitest/reporters/redis_reporter.rb (hunk 2 above)
Comment (casperisfine) [inline lib/minitest/reporters/redis_reporter.rb]: Nope. Minitest 💩

**Comment by casperisfine** [comment]
@wvanbergen Should be fixed.

**Comment by wvanbergen** [comment]
Ship!

**Pull request merged**